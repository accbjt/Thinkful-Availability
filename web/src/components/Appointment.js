import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const Appointment = props => {
  const { time, advisorId, bookAppointment } = props;

  return (
    <li key={time}>
      <time dateTime={time} className="book-time">
        {moment(time).format('MM/DD/YYYY h:mm a')}
      </time>
      <button
        className="book btn-small btn-primary m-4"
        type="submit"
        onClick={e => bookAppointment(e, { time, advisorId })}
      >
        Book
      </button>
    </li>
  );
};

Appointment.propTypes = {
  time: PropTypes.string.isRequired,
  advisorId: PropTypes.string.isRequired,
  bookAppointment: PropTypes.func.isRequired,
};

export default Appointment;
