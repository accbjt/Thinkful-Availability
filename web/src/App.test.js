import React from 'react';
import { shallow } from 'enzyme';

import Appointment from './components/Appointment';

it('renders without crashing', () => {
  const time = '2019-08-24T10:30:00-04:00';
  const component = shallow(
    <Appointment time={time} advisorId="1234" bookAppointment={() => {}} />,
  );

  expect(component).toMatchSnapshot();
});
