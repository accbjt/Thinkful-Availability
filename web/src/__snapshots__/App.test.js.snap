// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<li
  key="2019-08-24T10:30:00-04:00"
>
  <time
    className="book-time"
    dateTime="2019-08-24T10:30:00-04:00"
  >
    08/24/2019 7:30 am
  </time>
  <button
    className="book btn-small btn-primary m-4"
    onClick={[Function]}
    type="submit"
  >
    Book
  </button>
</li>
`;
